tests/optimize/test_backtest_detail.py

```diff
@@ -489,12 +489,12 @@
 # Test 31: trailing_stop should be triggered immediately on trade open candle.
 # stop-loss: 10%, ROI: 10% (should not apply)
 tc31 = BTContainer(data=[
-    # D   O     H     L     C    V    B  S
-    [0, 5000, 5050, 4950, 5000, 6172, 1, 0],
-    [1, 5000, 5500, 5000, 4900, 6172, 0, 0],    # enter trade (signal on last candle) and stop
-    [2, 4900, 5250, 4500, 5100, 6172, 0, 0],
-    [3, 5100, 5100, 4650, 4750, 6172, 0, 0],
-    [4, 4750, 4950, 4350, 4750, 6172, 0, 0]],
+    # D   O     H     L     C    V    B  S  SN
+    [0, 5000, 5050, 4950, 5000, 6172, 1, 0, ''],
+    [1, 5000, 5500, 5000, 4900, 6172, 0, 0, ''],    # enter trade (signal on last candle) and stop
+    [2, 4900, 5250, 4500, 5100, 6172, 0, 0, ''],
+    [3, 5100, 5100, 4650, 4750, 6172, 0, 0, ''],
+    [4, 4750, 4950, 4350, 4750, 6172, 0, 0, '']],
```

Comment:
I'm not certain why these changes are necessary.
i understand the need to add this "in general" to have it tested as well (That's `tc33`) - but i don't think it shold be needed to have it as "empty" columns - as the logic should be able to handle "not-assignment" of the buy-tag, too.